### ISSUE TITLE
```i18n:sync``` keep creating new folder for languages

### ISSUE THREAD

**Issue by NicolasRenault** [Alexandre-Fernandez/astro-i18n#36]
Hi,

I'm new to Astro and astro-i18n.
Is it normal that each time i run the commande ```i18n:sync``` the folder fr keep incrementing with more fr folders ?

![image](https://github.com/Alexandre-Fernandez/astro-i18n/assets/59968905/00b62831-8d86-4d78-9b9f-61dc5866723d)

Here is my astro.i18n.config.ts
![image](https://github.com/Alexandre-Fernandez/astro-i18n/assets/59968905/39530879-c854-4e5b-b5ac-3fe3ba91d2b2)

I have read the documentation and I don't understand where the issue come from.

Versions : 
"astro": "^2.4.1",
"astro-i18n": "^1.8.1",

**Comment by NicolasRenault** [comment]
Yeah sorry.

The repo is on my [github](https://github.com/NicolasRenault/portfolio). 
If the mistake come from me please don't waste your time fixing it and just let me know so I can fix it by myself.

Thanks for the quick answer !

**Comment by Alexandre-Fernandez** [comment]
If you want your routes translated you need to have `routeTranslations` in your config and not `translations`.
```ts
routeTranslations: {
		fr: {
			test: {
				welcome: "Titre",
			},
			about: {
				title: "A propos",
			},
		},
	}
```

**Comment by NicolasRenault** [comment]
Ok so I don't know if I understand right the notion of ```routeTranslations``` and ```translations```. Because for me route is gonna be the path in the URL. Is that write ?

I'm on a Windows 10 Pro 22h2 rn. 
Did you tried to run a second time the command ```npm run i18n:sync``` ?

**Comment by Alexandre-Fernandez** [comment]
Yes `routeTranslations` are for the url but that's what used to generate folders/pages. I actually ran it about 5 times without any issue. 
However I'm on linux and I don't have windows machine so that may be the problem.

**Comment by NicolasRenault** [comment]
Ok i'll give ```routeTranslations``` a try tommorow. I'll keep you in touch if the problem is fixed.

I'll try aswell to reproduce my problem on an other system.

**Comment by NicolasRenault** [comment]
Hi, 

I have looked at the documentation for ```routeTranslations``` and it seems that I want to use ```translations``` since I don't want to translate my URL but I want to translate text on the page in general. But this is off-topic I believe.

As said I have tried to reproduce the problem on another computer : 
- On my work computer, on a Windows Pro as well but 21H2 instead of 22H2
- On [Codesandbox](https://codesandbox.io/p/sandbox/astro-i18n-hj9mjb) which I believe runs on a Linux distro. To try with CodeSandbox you need to connect to it with your Github Account from the main page https://codesandbox.io/?from-app=1 . 
You will need to give me your address mail or the codesandbox's display name linked with the account so I can grant you access if you want to be able to launch a new terminal and try it by yourself.

The problem seems to come from Windows, at least for now. This can come from admins rights maybe?

**Comment by Alexandre-Fernandez** [comment]
> The problem seems to come from Windows, at least for now. This can come from admins rights maybe?
Don't know, if you want to contribute, you can download the library locally add some `console.log`, `npm build && npm pack` it and install it on a local astro project to see where the problem comes from.
The CLI sync logic starts [here](https://github.com/Alexandre-Fernandez/astro-i18n/blob/main/src/cli/commands/sync.ts).
I mainly maintain this library for my use-cases so I won't fix this unless I know exactly what and where the problem is.

**Comment by KRSogaard** [comment]
I am also having the issue of infinite translation directories on windows 11.
![image](https://github.com/Alexandre-Fernandez/astro-i18n/assets/11740656/3a2a9d30-5d1e-4ab4-a8db-1894f97d3f11)

**Comment by khmyznikov** [comment]
> I am also having the issue of infinite translation directories on windows 11. ![image](https://user-images.githubusercontent.com/11740656/241585715-3a2a9d30-5d1e-4ab4-a8db-1894f97d3f11.png)

Same

**Comment by NicolasRenault** [comment]
For now I have add a new command in my ```package.json```:

```json
"i18n:reset": "rm -rf src/pages/fr/"
```

> **Warning**: If you want to use this temporary fix make sure to add the right path for your specific languages and to run this command only at the root of your project !

As said in a previous comment I'll try to fix this issue but I'm running out of time for the moment.
If you want to join forces on this don't hesitate to contact me !

**Comment by NicolasRenault** [comment]
Hi @Alexandre-Fernandez, I have seen that you're currently working on the v2 of the library.
Do you plan to fix this issue in the new version? 
I can if you want help in the development to fix this with you on the v2.

**Comment by Alexandre-Fernandez** [comment]
> Hi @Alexandre-Fernandez, I have seen that you're currently working on the v2 of the library. Do you plan to fix this issue in the new version? I can if you want help in the development to fix this with you on the v2.

v2 is a complete rewrite and hopefully it will fix every issue with the current version. It will also add many features and simplify a lot of things.
I'm coding the v2 to be OS agnostic but because I do not use or have a windows machine to test things on I cannot be sure that it will work.
astro-i18n will only support linux and node officially, the rest is up to the community to contribute.

**Comment by NicolasRenault** [comment]
Great, I'll wait a bit a take a look at the sync when you are done with the rewrite for this part. I'll do tests and if needed try to do what's needed to fix potential issues.
Let me know if you need someone to test for Windows. I can give you my discord if needed, since we are both French it could be easier.

Good luck with the rewrite!

**Comment by Alexandre-Fernandez** [comment]
I just finished the new page generation :
https://github.com/Alexandre-Fernandez/astro-i18n/blob/2.0/astro-i18n/src/core/cli/commands/generate-pages.command.ts
You can test it, the new dev environment is easy to setup. 
To test it, clone and checkout to `2.0`, `pnpm install`, go to `apps/astro/src/pages` and then `pnpm generate:pages`, check that everything is ok.

**Comment by Alexandre-Fernandez** [comment]
This should be fixed in 2.0, as said no way to test it on windows. I'll reopen if needed.

**Comment by NicolasRenault** [comment]
Hi, sorry I wasn't able to test last week.

I just tested on my maching and it seems to be working !
Thanks you for your work.
